lib/faker/travel/train_station.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Faker
+  class Travel
+    class TrainStation < Base
+      class << self
+        ##
+        # Produces random Train Station by name and takes optional arguments for region and type
+        #
+        # @param region [String] Train station region: germany, spain, united_kingdom, united_states
+        #
+        # @param type [String] Train station type: metro, railway
+        #
+        # @return [String]
+        #
+        # @examples
+        # Faker::Travel::TrainStation.name(region: 'united_kingdom', type: 'metro') => "Brockley"
+        # Faker::Travel::TrainStation.name(type: 'railway') => "Düsseldorf Hauptbahnhof"
+        # Faker::Travel::TrainStation.name(region: 'spain') => "Santa Eulàlia"
+        #
+        # @faker.version next
+        def name(region: nil, type: nil)
+          regions = translate('faker.train_station').keys
+          types = translate('faker.train_station.united_kingdom').keys
+
+          if region.nil? && type.nil?
+            region = sample(regions)
+            type = sample(types)
+          elsif region.nil?
+            validate_arguments(type, types, 'type')
+            region = sample(regions)
+          elsif type.nil?
+            validate_arguments(region, regions, 'region')
+            type = sample(types)
+          end
+
+          fetch("train_station.#{region.downcase}.#{type.downcase}")
```

Comment:
it's nice that symbols and mixed-cased strings are easily supported by the generator, but at the same time I don't want people relying on that.

The reason is that if we support `United_Kingdom` from the start, we might end up needing to support `United Kingdom` or `united-kingdom` in the future, and similar things.

I think we should keep it simple for now and only offer support for the simple cases, changing this line to:
```suggestion
          fetch("train_station.#{region}.#{type}")
```